src/events.js

```diff
@@ -25,6 +26,33 @@ module.exports = function(Chart) {
 		return filteredEvents;
 	}
 
+	function applyHover(element, e, eventHandlers, beginHover) {
```

Comment:
I think it might be slightly cleaner to break this into two functions: one for starting hover and one for ending it